realm/realm-library/src/main/java/io/realm/RealmConfiguration.java

```
@@ -151,6 +159,23 @@ RealmProxyMediator getSchemaMediator() {
     }
 
     /**
+     * Returns input stream object to the Realm asset file.
+     *
+     * @return input stream to the asset file.
+     * @throws IOException if copying the file fails.
+     */
+    InputStream getAssetFile() throws IOException {
+        if (contextWeakRef != null) {
+            Context context = contextWeakRef.get();
+            if (context != null) {
+                return context.getAssets().open(assetFilePath);
+            }
+        }
+
+        return null;
```

See above. Throw exception instead of failing silently.